lib/webauthn/attestation.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require "cbor"
+require "openssl"
+require "webauthn/attestation_statement"
+require "webauthn/authenticator_data"
+
+module WebAuthn
+  class Attestation
+    def self.deserialize(attestation_object)
+      from_map(CBOR.decode(attestation_object))
+    end
+
+    def self.from_map(map)
+      new(
+        authenticator_data: WebAuthn::AuthenticatorData.new(map["authData"]),
+        attestation_statement: WebAuthn::AttestationStatement.from(map["fmt"], map["attStmt"])
+      )
+    end
+
+    attr_reader :authenticator_data, :attestation_statement
+
+    def initialize(authenticator_data:, attestation_statement:)
+      @authenticator_data = authenticator_data
+      @attestation_statement = attestation_statement
+    end
+
+    def valid_attested_credential?
+      authenticator_data.attested_credential_data_included? &&
+        authenticator_data.attested_credential_data.valid?
+    end
+
+    def valid_attestation_statement?(client_data_hash)
+      @type, @trust_path = attestation_statement.valid?(authenticator_data, client_data_hash)
+    end
+
+    def trustworthy?
+      case type
+      when WebAuthn::AttestationStatement::ATTESTATION_TYPE_NONE
+        acceptable_types.include?('None')
+      when WebAuthn::AttestationStatement::ATTESTATION_TYPE_SELF
+        acceptable_types.include?('Self')
+      else
+        acceptable_types.include?(type) && root_store.verify(subject_certificate, intermediate_certificates)
+      end
+    end
```

Comment:
This "ivar dependency/coupling" (`@type` and `@trust_path`) between `#trustworthy?` and `#valid_attestation_statement?` methods smells bad...

Keeping like this for now to keep backwards compatibility on how `AuthenticatorAttestationResponse` works as of today.

Reply:
I've been poking at this as well but took a slightly different path, as mentioned in https://github.com/cedarcode/webauthn-ruby/issues/294#issuecomment-569967671 by pushing in the the various formats to be checked before returning the trust path to be stored by the application. Just pushed https://github.com/cedarcode/webauthn-ruby/pull/304, have a look at the last commit.

Reply:
Cool, will take a look.

I am not going specifically in any path here. Just moved this almost verbatim into this new `Attestation` object.